fe/src/main/java/org/apache/doris/analysis/InsertStmt.java

```diff
@@ -397,40 +448,37 @@ private void analyzePlanHints(Analyzer analyzer) throws AnalysisException {
             }
         }
     }
-
-    private Expr checkTypeCompatibility(Column col, Expr expr) throws AnalysisException {
-        // TargeTable's hll column must be hll_hash's result
-        if (col.getType().equals(Type.HLL)) {
-            final String hllMismatchLog = "Column's type is HLL,"
-                    + " SelectList must contains HLL or hll_hash function's result, column=" + col.getName();
-            if (expr instanceof SlotRef) {
-                final SlotRef slot = (SlotRef) expr;
-                if (!slot.getType().equals(Type.HLL)) {
-                    throw new AnalysisException(hllMismatchLog);
-                }
-            } else if (expr instanceof FunctionCallExpr) {
-                final FunctionCallExpr functionExpr = (FunctionCallExpr) expr;
-                if (!functionExpr.getFnName().getFunction().equalsIgnoreCase("hll_hash")) {
-                    throw new AnalysisException(hllMismatchLog);
-                }
-            } else {
+    private void checkHllCompatibility(Column col, Expr expr) throws AnalysisException {
+        final String hllMismatchLog = "Column's type is HLL,"
```

Comment:
Why hll's compatibility check is s separated from normal compatibility check?

Reply:
Because HLL column will replace with SlotRef in Planner with a `varchar` type, and this will fail in this function. So we need check HLL column here.